backend/data/processors/courses_processing.py

```diff
@@ -123,7 +124,8 @@ def process_enrolment_path(processed: dict, formatted: dict, data: dict) -> None
             if "path_to" not in data[prereq]:
                 # Set up dict if not yet added
                 data[prereq]["path_to"] = dict()
-            data[prereq]["path_to"][processed["code"]] = 1
+            # data[prereq]["path_to"][processed["code"]] = 1
+            data[prereq]["path_to"] = courses.coursesUnlockedWhenTaken(prereq)
```

Comment:
This is missing userData parsed in, but where would this come from?

Reply:
yea - the way you have it now, it cant come from anywhere - so we should completely seperate this functionality from getCourse as you have mostly already done.